ext/src/ruby_api/execution_error.rs

```diff
@@ -1,35 +1,85 @@
-use crate::helpers::rhash_with_capacity;
+use crate::helpers::{rhash_with_capacity, TypedFrozenRArray};
 
 use super::root;
 use magnus::{
-    function, method,
+    function, gc, method,
     rb_sys::AsRawValue,
     scan_args::scan_args,
     typed_data::{self, Obj},
-    Error, Module, Object, RHash, Value,
+    DataTypeFunctions, Error, Integer, Module, Object, RArray, RHash, TypedData, Value,
 };
 use std::borrow::Cow;
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-#[magnus::wrap(class = "Bluejay::ExecutionError")]
+#[magnus::wrap(class = "Bluejay::ExecutionError::ErrorLocation")]
+pub struct ErrorLocation {
+    line: usize,
+    column: usize,
+}
+
+impl ErrorLocation {
+    pub fn new(line: usize, column: usize) -> Self {
+        Self { line, column }
+    }
+
+    fn rb_new(line_v: Integer, col_v: Integer) -> Result<Self, Error> {
+        let line = line_v.try_convert::<usize>().unwrap();
+        let column = col_v.try_convert::<usize>().unwrap();
+        Ok(Self::new(line, column))
+    }
```

Comment:
Could we use keyword args here, something like this?
```rust
    fn rb_new(kw: RHash) -> Result<Self, Error> {
        let (line, column) =
            get_kwargs::<_, (usize, usize), (), ()>(kw, &["line", "column"], &[])?.required;
        Ok(Self { line, column })
    }
```

Reply:
Nice, done!